src/numbuf/python/src/pynumbuf/numbuf.cc

```diff
@@ -275,12 +345,9 @@ static PyObject* store_list(PyObject* self, PyObject* args) {
   }
   CHECK(error_code == PlasmaError_OK);
 
-  auto target = std::make_shared<FixedBufferStream>(LENGTH_PREFIX_SIZE + data, size);
-  std::shared_ptr<arrow::ipc::FileWriter> writer;
-  ipc::FileWriter::Open(target.get(), batch->schema(), &writer);
-  writer->WriteRecordBatch(*batch, true);
-  writer->Close();
-  *((int64_t*)data) = size;
+  auto target = std::make_shared<FixedBufferStream>(LENGTH_PREFIX_SIZE + data, total_size);
```

Comment:
I added a variant of this class (hopefully suitable for your needs) recently to Arrow: https://github.com/apache/arrow/blob/master/cpp/src/arrow/io/memory.h#L72